### ISSUE TITLE
FR: Add methods for removing tokens

### ISSUE THREAD

**Issue by steventrouble** [rustformers/llm#332]
**tl;dr:** Add a `delete_tokens(num: usize)` method to `InferenceSession`.

## Motivation

The session architecture is brilliant for reducing computions! 

While it currently only works in a feed-forward manner, it would be great to delete tokens too. That way users can more easily implement an "undo" button in an app without having to deal with tons of state snapshots.

This would enable `llm` to be used in things like smartphone keyboard autocomplete, which requires super fast inference and almost always deals with the same text with some minor changes at the end of it.

*Implementation proposal forthcoming*

**Comment by philpax** [comment]
This would be great! I'm not sure how possible it is - the memory K and V tensors are mutated by the evaluation process, and I'm not sure if the changes are reversible. I encourage you to investigate, though!
